src/lib/components/CommunitySelectionModal/CommunitySelectionModal.js

```diff
@@ -0,0 +1,250 @@
+// This file is part of React-Invenio-Deposit
+// Copyright (C) 2020 CERN.
+//
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from "react";
+import {
+  Header,
+  Grid,
+  Segment,
+  Button,
+  Modal,
+  Item,
+  Menu,
+  Container,
+  Icon,
+} from "semantic-ui-react";
+import {
+  ReactSearchKit,
+  SearchBar,
+  ResultsLoader,
+  EmptyResults,
+  Error,
+  InvenioSearchApi,
+  ResultsList,
+  Pagination,
+} from "react-searchkit";
+import { OverridableContext } from "react-overridable";
+import { Formik } from "formik";
+import { CommunityListItem } from "./CommunityListItem";
+import PropTypes from "prop-types";
+import { Trans, i18next } from "@translations/i18next";
+
+const overriddenComponents = {
+  "communities.ResultsList.item": CommunityListItem,
+};
+
+export class CommunitySelectionModal extends Component {
+  constructor(props) {
+    super(props);
+
+    this.apiEndpoints = {
+      allCommunities: "https://127.0.0.1:5000/api/communities",
+      myCommunities: "https://127.0.0.1:5000/api/user/communities",
+    };
+
+    const defaultEndpoint = this.apiEndpoints.allCommunities;
+
+    this.state = {
+      modalOpen: false,
+      selectedEndpoint: defaultEndpoint,
+    };
+  }
+
+  onSave = (chosenValues, resetFormCallBack) => {
+    const { onCommunityChange } = this.props;
+    const { chosenCommunity } = chosenValues;
+
+
+    onCommunityChange(chosenCommunity);
+    this.setState({modalOpen: false}, resetFormCallBack)
+
+  };
+
+  render() {
+    const { modalOpen, selectedEndpoint } = this.state;
+    const { allCommunities, myCommunities } = this.apiEndpoints;
+    const { chosenCommunity, trigger } = this.props;
+
+    const searchApi = new InvenioSearchApi({
+      axios: { url: selectedEndpoint },
+    });
+
+    const searchbarPlaceholder =
+      selectedEndpoint === allCommunities
+        ? i18next.t("Search in all communities")
+        : i18next.t("Search in my communities");
+
+    return (
```

Comment:
Can you extract some of the inner parts in separate cmps, in this same file? It will help readibility.